flatfs/flatfs.go

```diff
@@ -47,21 +46,19 @@ func New(path string, prefixLen int, sync bool) (*Datastore, error) {
 		return nil, ErrBadPrefixLen
 	}
 	fs := &Datastore{
-		path: path,
-		// convert from binary bytes to bytes of hex encoding
-		hexPrefixLen: prefixLen * hex.EncodedLen(1),
-		sync:         sync,
+		path:      path,
+		prefixLen: prefixLen,
+		sync:      sync,
 	}
 	return fs, nil
 }
 
-var padding = strings.Repeat("_", maxPrefixLen*hex.EncodedLen(1))
+var padding = strings.Repeat("_", maxPrefixLen)
 
 func (fs *Datastore) encode(key datastore.Key) (dir, file string) {
-	safe := hex.EncodeToString(key.Bytes()[1:])
-	prefix := (safe + padding)[:fs.hexPrefixLen]
+	prefix := (key.String() + padding)[:fs.prefixLen]
 	dir = path.Join(fs.path, prefix)
-	file = path.Join(dir, safe+extension)
+	file = path.Join(dir, key.String()+extension)
```

Comment:
And `path.Join(dir, key.String()+extension)` will include the '/' from the datastore key.